packages/react/src/components/hooks/useLayouts.ts

```diff
@@ -0,0 +1,32 @@
+import { Video, VideoPositions } from "@signalwire/js";
+import { useEffect, useState } from "react";
+
+export default function useLayouts(roomSession: Video.RoomSession | null) {
+  const [layouts, setLayouts] = useState<string[]>([]);
+  const [currentLayout, setCurrentLayout] = useState("");
+  useEffect(() => {
+    if (roomSession === null || roomSession === undefined) return;
+    async function onRoomJoined(room: any) {
+      let layout_list = (await roomSession?.getLayouts())?.layouts;
+      setLayouts(layout_list || []);
+      setCurrentLayout(room.room_session.layout_name);
+    }
+    async function onLayoutChanged(e: any) {
+      setCurrentLayout(e.layout.name);
+    }
+    roomSession.on("room.joined", onRoomJoined);
+    roomSession.on("layout.changed", onLayoutChanged);
+
+    return () => {
+      roomSession.off("layout.changed", onLayoutChanged);
+      roomSession.off("layout.changed", onLayoutChanged);
+    };
+  }, [roomSession]);
+  return {
+    layouts,
+    setLayout(name: string, positions?: VideoPositions) {
+      roomSession?.setLayout({ name, positions });
+    },
```

Comment:
let's expose the `setLayout` method with the same interface of `RoomSession.setLayout`, so that if the signature changes we don't have to go back here and update it.